### ISSUE TITLE
discord: Discord with OpenASAR fails to show title bar

### ISSUE THREAD

**Issue by Ringlings** [NixOS/nixpkgs#237129]
### Describe the bug
When using Discord from nixpkgs with OpenASAR overridden on top the title bar is missing

### Steps To Reproduce
1. Install Discord with OpenASAR enabled
2. Launch Discord
3. Title bar is missing

### Expected behavior
The window has a GTK title bar

### Screenshots
![image](https://github.com/NixOS/nixpkgs/assets/110815694/6d57079a-4797-437f-adba-7a5f41257a05)

### Additional context
Using GNOME 44.1 under Wayland

### Notify maintainers
@MP2E @Scrumplex @artturin @infinidoge @jopejoe1

### Metadata
```console
[ringlings@Adylis:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.31, NixOS, 23.05 (Stoat), 23.05.730.85bcb95aa83`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"home-manager-23.05.tar.gz, nixos-23.05, nixos-unstable, nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

**Comment by Infinidoge** [comment]
That is... very strange. This feels more like a compositor-side issue, since I don't think Discord itself handles the title bar under Wayland(?).

Can you confirm that it properly has the title bar when OpenASAR is disabled?

**Comment by Ringlings** [comment]
Yep, we've tried just discord, discord w/ vencord, discord w/ OA, and discord w/ both.
The issue only occurs when OA is enabled

**Comment by surfaceflinger** [comment]
It's caused by your environment variables https://github.com/Ringlings/nixos/blob/master/patches/wayland.nix

Discord's electron setup simply isn't ready for wayland yet

**Comment by getchoo** [comment]
i'm able to reproducible this currently on nixos unstable with mutter. the vanilla client seems to be fine when using wayland now, and this only appears when openasar is enabled

you can try using these two commands:

```
$ NIXPKGS_ALLOW_UNFREE=1 nix run --impure --expr \
      'with import (builtins.getFlake "nixpkgs/nixos-unstable") {};
        discord-canary.override {withOpenASAR = true;}
      '

$ NIXPKGS_ALLOW_UNFREE=1 nix run --impure --expr \
      'with import (builtins.getFlake "nixpkgs/nixos-unstable") {};
        discord-canary                                
      '
```

these (should) result in the following
![image](https://github.com/NixOS/nixpkgs/assets/48872998/cf3a63a9-dedf-4d83-8c15-3dfc524d96c7)

![image](https://github.com/NixOS/nixpkgs/assets/48872998/8fc0f366-af05-4a86-b210-870e8f74bfcd)
